src/OpenTelemetry.Exporter.Jaeger/Implementation/EnumerationHelper.cs

```diff
@@ -0,0 +1,197 @@
+﻿// <auto-generated>
+// <copyright file="EnumerationHelper.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+using System.Reflection.Emit;
+
+namespace OpenTelemetry.Exporter.Jaeger.Implementation
+{
+    internal class DictionaryEnumerator<TKey, TValue, TState> : Enumerator
```

Comment:
Could we move this into a shared library so other places can also take advantage? Moving it alongside the `EvictingQueue` would probably work.

Reply:
@MikeGoldsmith I thought about doing that but I was a little hesitant because to share it we would really need it to be public in OpenTelemetry dll which would then expose it to consumers, which felt wrong. Maybe it would be better just to link into other OT projects as needed? Based on how the child IEnumerables of structs (Attributes, Links, Links.Attributes, etc) live off the parent (also struct) SpanData, most exporters will probably need this optimization to iterate without boxing. LMK what you think.